### PR TITLE
Basic implementation of an LCM kernel to address #261

### PR DESCRIPTION
**Pull request by rajkumarkarthik**
This code creates a simple wrapper for an LCM kernel by adding multiple `MultitaskKernel` objects. This addresses the issue raised in #261. However, the code throws a `RuntimeError` that I don't fully understand. Does anyone know what is going on?

cc @Balandat (who contributed to this PR)

### CODE REVIEW

**Comment by jacobrgardner** [comment]
Do you have any example code or an exact error message?

**Comment by rajkumarkarthik** [comment]
Yes, here is a notebook that shows the error. (Github won't let me upload a .ipynb file, so just remove the .txt extension when you use it.)

[Multitask_GP_Regression.ipynb.txt](https://github.com/cornellius-gp/gpytorch/files/2398746/Multitask_GP_Regression.ipynb.txt)

File: gpytorch/kernels/lcm_kernel.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from gpytorch.kernels.multitask_kernel import MultitaskKernel
+
+
+class LCMKernel(Kernel):
+    """
+    This kernel supports the LCM kernel.
+    """
+    def __init__(self, base_kernels, n_tasks, rank=1, task_covar_prior=None):
+        """
+        Args:
```
Comment (eytan) [inline gpytorch/kernels/lcm_kernel.py]: a little light on the docstrings

**Comment by jacobrgardner** [comment]
Fixed the issue you were having. Part of the multitask refactor we did to add them as first class objects was to require kernels to implement a `size(x1, x2)` method if they anticipate returning covariance matrices that have a size different from `x1.size(1) x x2.size(1)` (where x1, x2 are `b x n x d`).

Since all multitask kernels return lazy tensors that represent matrices of size `nt x nt` rather than just `n x n`, our internals need to know about that.

**Comment by jacobrgardner** [comment]
Can we change the base branch for this to beta or do you need this in master?

**Comment by rajkumarkarthik** [comment]
I've been working on the branch `create_lcm_wrapper`. Did this push to master?

**Comment by jacobrgardner** [comment]
Nope you're good! I meant can you modify the PR so that it merges in to beta instead of master

**Comment by jacobrgardner** [comment]
I went ahead and updated it -- main thing is just to make sure everything is compatible with everything else going in to beta.

**Comment by rajkumarkarthik** [comment]
Thanks! What is the difference between master and beta again? Is beta for features we don't want to merge into master right away?

**Comment by jacobrgardner** [comment]
`beta` is the base branch for the beta release we are hoping to do shortly -- once everything is ready, everything we're doing for the beta will get merged in to master in one go. 

Since `beta` is quite a bit ahead of master, I just figured it would be useful to make sure everything you're doing is compatible with everything that has already gone on the `beta` branch. Looks like the checks did pass, so should be all good on that aspect.

**Comment by Balandat** [comment]
This looks good to me - would you mind supplying some basic unit tests for this?

**Comment by rajkumarkarthik** [comment]
Done! This test does a simple sanity check if LCM and ICM provide the same results in the case where only one base kernel is provided for LCM.

File: test/examples/test_lcm_kernel_regression.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from gpytorch.means import ConstantMean, MultitaskMean
+from gpytorch.likelihoods import MultitaskGaussianLikelihood
+from gpytorch.random_variables import MultitaskGaussianRandomVariable
+
+
+class MultitaskGPModel(gpytorch.models.ExactGP):
+    def __init__(self, train_x, train_y, likelihood):
+        super(MultitaskGPModel, self).__init__(train_x, train_y, likelihood)
+        self.mean_module = MultitaskMean(ConstantMean(), n_tasks=2)
+        self.base_kernel_list = [RBFKernel()]
```
Comment (Balandat) [inline test/examples/test_lcm_kernel_regression.py]: Why make this an attribute of the instance? This shouldn't be necessary.

File: test/examples/test_lcm_kernel_regression.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        mean_x = self.mean_module(x)
+        covar_x = self.covar_module(x)
+        return MultitaskGaussianRandomVariable(mean_x, covar_x)
+
+
+class MultitaskGPModel_ICM(gpytorch.models.ExactGP):
+    def __init__(self, train_x, train_y, likelihood):
+        super(MultitaskGPModel_ICM, self).__init__(train_x, train_y, likelihood)
+        self.mean_module = MultitaskMean(ConstantMean(), n_tasks=2)
+        self.base_kernel = RBFKernel()
```
Comment (Balandat) [inline test/examples/test_lcm_kernel_regression.py]: same here, no need to add this, this should be chained together automatically.

**Comment by Balandat** [comment]
Thanks for working on this!

**Pull request merged**